### ISSUE TITLE
Add legend to grid view

### ISSUE THREAD

**Issue by bbovenzi** [apache/airflow#28737]
### Description

It would be useful to add a simple legend to the grid view.

- Help a user understand the icons for a manual / backfill / dataset triggered dag runs
- Help a user understand the symbol for tasks/runs with a note

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
> @bbovenzi is this issue taken? I would like to try my hands here given some more reference material here.

Nope! It's all yours. Feel free to ask any questions here.

We'll probably want to make a new react component and put add it in the main grid view component [here](https://github.com/apache/airflow/blob/main/airflow/www/static/js/dag/grid/index.tsx).

- We'll want to refer to each [RunTypeIcon](https://github.com/apache/airflow/blob/main/airflow/www/static/js/components/RunTypeIcon.tsx)

- And we'll want to explain the [gradient](https://github.com/apache/airflow/blob/main/airflow/www/static/js/utils/index.ts#L128) we use to indicate if there is a note

**Comment by amoghrajesh** [comment]
Hi, @bbovenzi I am new to Airflow and would likewith  to get a few things clarified before starting.
- Do we have some kind of legends in the airflow codebase that i can refer to? Also, are we talking about legend being a small component that gives out some information like in graphs etc or react-legend?(https://www.npmjs.com/package/react-legend)
- I would appreciate it if you could provide me some online reference for the same

**Comment by bbovenzi** [comment]
You can refer to the legend row of dag run/task instance statuses. Basically, just a representation of each icon/gradient with a word or two next to it.
We don't need an extra mom package. Just a simple react component.

**Comment by amoghrajesh** [comment]
@bbovenzi a follow-up on which row you meant here? Can you show me where the legend is here? 
![image](https://user-images.githubusercontent.com/35884252/211298924-2a6105cf-e993-4073-bfe0-f8ad1368bfa0.png)

**Comment by ashb** [comment]
@amoghrajesh That is not the Gird view we are talking about -- this one is https://airflow.apache.org/docs/apache-airflow/stable/ui.html#grid-view

**Comment by amoghrajesh** [comment]
@ashb @bbovenzi coming back to this issue after some time. We are trying to add some legend to this page if I am not mistaken:
![image](https://user-images.githubusercontent.com/35884252/220895861-296c82da-5514-401d-a136-a01d6599daa1.png)

1. We can add a small legend for type of run
2. A small legend for tasks vs runs? How do I identify them in terms of logos?

**Comment by Grzyblon** [comment]
@bbovenzi is this issue still relevant?

**Comment by Grzyblon** [comment]
![image](https://github.com/user-attachments/assets/a21d2f6a-3699-44a6-bfab-2dfbc1d43416)

Is that what you have in mind ?
I haven't figured out yet how should the icon looks in point about a DAG runs with a note.

**Comment by bbovenzi** [comment]
Something like that in a tooltip, yes! For notes, let's make a mock task instance and render the StatusWithNotes component

**Comment by Grzyblon** [comment]
I've moved that list to DagRunTooltip.
![DAG_run_types_in_tooltip_airflow](https://github.com/user-attachments/assets/6aeaebcc-3495-4572-aa63-a46bb8c794f6)

I was considering putting that in a tooltip but I had an impression that it should be easily visible. That's why I put it first under the grid. Also tooltip already have information about run type (maybe we should add an icon like run type description in Details tab). Finally maybe it would be nice to highlight DAG runs of given type after hovering mouse on list item just like task statuses. Of course same with note.

**Comment by bbovenzi** [comment]
No, definitely not in every single dag run tooltip. Once a user learns the icons, they don't need to be reminded every single time.

But maybe we should add those icons to the run_type filters.
<img width="190" alt="Screenshot 2024-07-30 at 2 44 50 PM" src="https://github.com/user-attachments/assets/a4032f3e-d618-4fd3-b06f-2d9c88f637b0">

**Comment by Grzyblon** [comment]
@bbovenzi  I’ve prepared a few proposals based on your comments.

> But maybe we should add those icons to the run_type filters.

It was not so trivial as I thought it'll be. Here are results.
![DAG_run_types_in_filter](https://github.com/user-attachments/assets/af6755cc-430d-45d0-b8cf-b7fadf7c13f8)
I’ve also added rendering StatusWithNotes next to a “Contains a note” text with the same status color as task instance/ DAG run tooltip originates from.
![DAG_run_typeicon_in_tooltip_airflow](https://github.com/user-attachments/assets/14bf6f61-74dd-42e6-9189-d1b7c520761d)

> Just an InfoTooltip component is fine I think.

I’ve also made grid legend using InfoTooltip component placed over Grid component. This definitely needs some more tweaks.
![DAG_run_types_in_tooltip_airflow](https://github.com/user-attachments/assets/75103e77-55f9-4f1c-bd8e-3b79da7ff82e)

In my opinion adding icons to run type options in filter plus rendering StatusWithNotes component in InstanceTooltip and DagRunTooltip should be sufficient. That way user will be getting familiar with the graphics gradually as they use these features. In that case adding InfoTooltip with whole legend seems redundant.

Is that looks good?

**Comment by bbovenzi** [comment]
looking good! we may need to adjust the statuswith notes margins but otherwise its great!

**Comment by Grzyblon** [comment]
Great! I’ll be unavailable for next few days now. I’ll tweaks things a bit, clean up code and prepare PR when I get back.

**Comment by RoyLee1224** [comment]
Hi @Grzyblon, are you still planning to work on this issue? I saw your last update was a while ago. If you're no longer available, I would be happy to take it from here.

cc @bbovenzi

**Comment by bbovenzi** [comment]
Good call. Assigned to you now @RoyLee1224

**Comment by uranusjr** [comment]
Are you still working on this @RoyLee1224?

**Comment by RoyLee1224** [comment]
@uranusjr The issue needs to be redone after the gridTI's Tooltip is modified. 

I don't have time for it at the moment; it can be left for someone else, or maybe wait until I've finished my other issues.
